flows/urns.go

```diff
@@ -27,8 +28,78 @@ func ValidateURNScheme(fl validator.FieldLevel) bool {
 	return urns.IsValidScheme(fl.Field().String())
 }
 
+type ContactURN struct {
+	urns.URN
+	channel Channel
+}
+
+func (u *ContactURN) Channel() Channel { return u.channel }
+
+// Resolve is called when a URN is part of an excellent expression
+func (u *ContactURN) Resolve(key string) interface{} {
+	switch key {
+	case "scheme":
+		return u.URN.Scheme()
+	case "path":
+		return u.URN.Path()
+	case "display":
+		return u.URN.Display()
+	case "channel":
+		return u.Channel()
+	}
+	return fmt.Errorf("no field '%s' on URN", key)
+}
+
+func (u *ContactURN) Default() interface{} { return u }
+
+func (u *ContactURN) String() string { return string(u.URN) }
+
 // URNList is the list of a contact's URNs
-type URNList []urns.URN
+type URNList []*ContactURN
+
+func ReadURNList(session Session, rawURNs []urns.URN) (URNList, error) {
+	l := make(URNList, len(rawURNs))
+
+	for u := range rawURNs {
+		scheme, path, query, display := rawURNs[u].ToParts()
+
+		// re-create the URN without the query component
+		queryLess, err := urns.NewURNFromParts(scheme, path, "", display)
+		if err != nil {
+			return nil, err
+		}
+
+		parsedQuery, err := url.ParseQuery(query)
```

Comment:
Occurs to me we may want to think about what the URN value that is exposed is.. gets kinda tricky, but ya, a URN like `facebook:12345#rowanseymour` is treated the same in our system as `facebook:12345#rowan`, the display portion is not part of the "identity" of the URN.

I'm not worried about us not doing the right thing here, but wondering about users using these URNs in outside systems and the like, do they have an easy way to get a URN `identity` as we would call it? (stripped of channel AND display)